base/subarray.jl

```diff
@@ -395,9 +395,9 @@ end
 # Since bounds-checking is performance-critical and uses
 # indices, it's worth optimizing these implementations thoroughly
 axes(S::SubArray) = (@_inline_meta; _indices_sub(S, S.indices...))
-_indices_sub(S::SubArray) = ()
-_indices_sub(S::SubArray, ::Real, I...) = (@_inline_meta; _indices_sub(S, I...))
-function _indices_sub(S::SubArray, i1::AbstractArray, I...)
+_indices_sub(@nospecialize(S::SubArray)) = ()
```

Comment:
Maybe this argument can just be removed completely? I don't see it used anywhere.

Reply:
The first argument to `reindex` also seems to be unused?

Reply:
I think it's just historical cruft.